indexing-service/src/test/java/io/druid/indexing/overlord/TaskLifecycleTest.java

```diff
@@ -273,24 +328,53 @@ public void setUp() throws Exception
     publishCountDown = new CountDownLatch(1);
     announcedSinks = 0;
     pushedSegments = 0;
-    tmp = Files.createTempDir();
+    tmpDir = temporaryFolder.newFolder();
 
     final TaskQueueConfig tqc = new DefaultObjectMapper().readValue(
-        "{\"startDelay\":\"PT0S\", \"restartDelay\":\"PT1S\"}",
+        "{\"startDelay\":\"PT0S\", \"restartDelay\":\"PT1S\", \"storageSyncRate\":\"PT0.5S\"}",
         TaskQueueConfig.class
     );
     indexSpec = new IndexSpec();
-    ts = new HeapMemoryTaskStorage(
-        new TaskStorageConfig(null)
-        {
-        }
-    );
+
+    if (taskStorageType.equals("HeapMemoryTaskStorage")) {
+      ts = new HeapMemoryTaskStorage(
+          new TaskStorageConfig(null)
+          {
+          }
+      );
+    } else if (taskStorageType.equals("MetadataTaskStorage")) {
+      MetadataStorageTablesConfig tablesConfig = MetadataStorageTablesConfig.fromBase("test");
+      testDerbyConnector = new TestDerbyConnector(
+          Suppliers.ofInstance(
+              new MetadataStorageConnectorConfig()
+          ),
+          Suppliers.ofInstance(
+              tablesConfig
+          )
+      );
+      mapper = new DefaultObjectMapper();
+      mapper.registerSubtypes(
+          new NamedType(MockExceptionalFirehoseFactory.class, "mockExcepFirehoseFactory"),
+          new NamedType(MockFirehoseFactory.class, "mockFirehoseFactory")
```

Comment:
```MockFirehoseFactory ``` and ```MockExceptionalFirehoseFactory``` already exists in the original code. I moved around it a bit so that it can be serialize/deserialize properly by JacksonJson